thirdparty/kyaml/runfn/runfn_test.go

```diff
@@ -1095,8 +671,35 @@ replace: StatefulSet
 	assert.Contains(t, string(b), "kind: Deployment")
 }
 
-// TestCmd_Execute_setOutput tests the execution of a filter reading and writing to a dir
-func TestCmd_Execute_setFunctionPaths(t *testing.T) {
+func getFnConfigPathFilterProvider(t *testing.T, r *RunFns) func(runtimeutil.FunctionSpec, *yaml.RNode, currentUserFunc) (kio.Filter, error) {
+	return func(f runtimeutil.FunctionSpec, node *yaml.RNode, currentUser currentUserFunc) (kio.Filter, error) {
+		// parse the filter from the input
+		filter := yaml.YFilter{}
+		b := &bytes.Buffer{}
+		e := yaml.NewEncoder(b)
+		var err error
+		if r.FnConfigPath != "" {
+			node, err = r.getFunctionConfig()
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+		if !assert.NoError(t, e.Encode(node.YNode())) {
+			t.FailNow()
+		}
+		e.Close()
+		d := yaml.NewDecoder(b)
+		if !assert.NoError(t, d.Decode(&filter)) {
+			t.FailNow()
+		}
+
+		return filters.Modifier{
+			Filters: []yaml.YFilter{{Filter: yaml.Lookup("kind")}, filter},
+		}, nil
+	}
+}
+
+func TestCmd_Execute_setFnConfigPath(t *testing.T) {
```

Comment:
are there e2e tests for imperative functions?

please update the tracker: https://docs.google.com/spreadsheets/d/1V5f1UpL0_WqJPyeFTaR0bR4acCGtoUENYHVvV_HOlrc/edit#gid=1609746542

Reply:
Note that having example packages is about more than e2e testing. They are also are examples we can provide end users as part of our documentation. We need at least one example package for each of the rows in that doc.

Reply:
Will do